contribs/drt/src/main/java/org/matsim/contrib/drt/routing/DrtRoute.java

```diff
@@ -39,44 +47,60 @@
  */
 public class DrtRoute extends AbstractRoute {
 	public final static String ROUTE_TYPE = TransportMode.drt;
-
-	private OptionalTime maxWaitTime = OptionalTime.undefined();
-	private OptionalTime directRideTime = OptionalTime.undefined();
+	private RouteDescription routeDescription = null;
 
 	public DrtRoute(Id<Link> startLinkId, Id<Link> endLinkId) {
 		super(startLinkId, endLinkId);
+		this.routeDescription = new RouteDescription();
 	}
 
 	public double getDirectRideTime() {
-		return directRideTime.seconds();
+		return routeDescription.getDirectRideTime();
 	}
 
 	public double getMaxWaitTime() {
-		return maxWaitTime.seconds();
+		return routeDescription.getMaxWaitTime();
+	}
+
+	public List<String> getUnsharedPath() {
+		return routeDescription.getUnsharedPath();
+	}
+
+	public double getMaxTravelTime() {
+		return getTravelTime().seconds(); // currently DrtRoute.travelTime is set to the max allowed travel time
 	}
 
 	public void setDirectRideTime(double directRideTime) {
-		this.directRideTime = OptionalTime.defined(directRideTime);
+		this.routeDescription.setDirectRideTime(directRideTime);
 	}
 
 	public void setMaxWaitTime(double maxWaitTime) {
-		this.maxWaitTime = OptionalTime.defined(maxWaitTime);
+		this.routeDescription.setMaxWaitTime(maxWaitTime);
 	}
 
-	public double getMaxTravelTime() {
-		return getTravelTime().seconds(); // currently DrtRoute.travelTime is set to the max allowed travel time
+	public void setUnsharedPath(VrpPathWithTravelData unsharedPath) {
+		List<String> links = new ArrayList<String>();
+		unsharedPath.iterator().forEachRemaining(l->{links.add(l.getId().toString());});
+		this.routeDescription.setUnsharedPath(links);
 	}
 
+
 	@Override
 	public String getRouteDescription() {
-		return maxWaitTime.seconds() + " " + directRideTime.seconds();
+		try {
+			return new ObjectMapper().writeValueAsString(routeDescription);
+		} catch (JsonProcessingException e) {
+			throw new RuntimeException(e);
+		}
 	}
 
 	@Override
 	public void setRouteDescription(String routeDescription) {
```

Comment:
@steffenaxer This is called when the route is read from a plans file. So here could be a condition to fall back to the old format in `routeDescription`.

Reply:
Ok, got it. I will provide a retrofit version